core/src/main/java/org/springframework/security/access/intercept/AbstractSecurityInterceptor.java

```diff
@@ -375,8 +379,25 @@ public boolean isValidateConfigAttributes() {
 
 	public abstract SecurityMetadataSource obtainSecurityMetadataSource();
 
+	/**
+	 * Sets the {@link AccessDecisionManager} instance.
+	 * @param accessDecisionManager the {@link AccessDecisionManager} instance
+	 * @deprecated use {@link #setAuthorizationManager(AuthorizationManager)} instead
+	 */
+	@Deprecated
 	public void setAccessDecisionManager(AccessDecisionManager accessDecisionManager) {
-		this.accessDecisionManager = accessDecisionManager;
+		this.authorizationManager = new AuthorizationManagerAdapter<>(accessDecisionManager);
+	}
+
+	/**
+	 * Sets the {@link AuthorizationManager} instance.
+	 * @param authorizationManager the {@link AuthorizationManager} instance
+	 * @throws IllegalArgumentException if an {@link AccessDecisionManager} is already set
+	 */
+	public void setAuthorizationManager(AuthorizationManager<?> authorizationManager) {
+		Assert.isTrue(!(this.authorizationManager instanceof AuthorizationManagerAdapter),
```

Comment:
In newer APIs Spring Security checks for null in the setter. Does that work here?